### DISCUSSION TITLE
2.3.3

### DISCUSSION THREAD

**Discussion by mashb1t** [mashb1t/Fooocus#31] [Announcements]
## What's Changed
* feat: support download of huggingface files from a mirror website
* feat: update interposer from v3.1 to v4.0
* feat: update animaPencilXL from 2.0.0 to 3.1.0
* feat: add button to reconnect UI without having to reload the page
* feat: add performance hyper-sd based on 4step LoRA
* feat: add random style checkbox
* feat: add VAE select
* feat: add full raw prompt to history log
* feat: add AlignYourStepsScheduler
* feat: add tcd sampler
* refactor: code cleanup

**Full Changelog**: https://github.com/mashb1t/Fooocus/compare/2.3.1...2.3.3

<hr /><em>This discussion was created from the release <a href='https://github.com/mashb1t/Fooocus/releases/tag/2.3.3'>2.3.3</a>.</em>

**Comment by eisenbruch** [comment]
I feel silly - how do I upgrade to this version? I installed using your windows instructions and the run.bat file

**Reply by mashb1t** [reply]
normally Fooocus should pull the latest version itself, but you may have to execute `git pull` manually in the Fooocus folder.
It may also be possible that there is the repository of lllyasviel linked, please check with `git origin -v`.

**Reply by eisenbruch** [reply]
ah yeah, it did load fooocus version 2.3.3 (bashb1t)

my bad :)

**Comment by gh-aam** [comment]
Thanks for the hard work!
"Random Style" is awesome and works perfectly. "Hyper-SD" is good enough. Translate option sometimes can't translate properly from Bangla to English. In performance section, instead of using details like "4 steps, intermediate results disabled" you could used tooltip/infotip (I don't know if it should be or not. It is just my point of view as I prefer compact UI.).

**Reply by mashb1t** [reply]
I prefer information given directly and tooltips are always problematic when using mobile devices. I've noticed that with the style preview images which are not optimal as tooltips, but good enough compared to overloading the UI, so also for performance this is the balance between too much information and visibility.

**Reply by gh-aam** [reply]
That's right! I got your point. I also use Fooocus from mobile device using Colab whenever I feel lazy. Enabling "#enable-accessibility-page-zoom" Chrome flag and setting the page zoom level to 80% makes the UI perfect (though sometimes two finger zooming is necessary).

**Comment by mald55** [comment]
How difficult will it be to make Fooocus work with SD3 when it comes out? Will you not know until then?

**Reply by mashb1t** [reply]
The paper has already been released, but the whole pipeline probably needs to be restructured and adjusted to allow SDXL, SD1.5 (Refiners) and SD3 in parallel without breaking anything. The good thing is that Fooocus in general can synergize with other open source projects such as Diffusers (code needs to be completely extracted) or ComfyUI (already parts of Comfy in Fooocus), so i assume as soon as one of the projects has working pipeline support the others will follow.

**Reply by E2GO** [reply]
D'you know how great you are?))

**Comment by kalle07** [comment]
i think iam not that up to date, but thit you know already?
https://github.com/MackinationsAi/SD3

may you think 1-2 month we dont need any api-key ?

**Reply by E2GO** [reply]
Just so you know)
https://x.com/Lykon4072/status/1791820252243800253
